crates/pallet-domains/src/lib.rs

```diff
@@ -468,6 +461,17 @@ mod pallet {
         OptionQuery,
     >;
 
+    /// The consensus block hash used to verify ER, only store the consensus block hash for a domain
+    /// if that consensus block contains bundle of the domain, the hash will be pruned when the ER
+    /// that point to the consensus block is pruned.
+    ///
+    /// TODO: this storage is unbounded in some cases, see https://github.com/subspace/subspace/issues/1673
+    /// for more details, this will be fixed once https://github.com/subspace/subspace/issues/1731 is implemented.
+    #[pallet::storage]
+    #[pallet::getter(fn consensus_hash)]
+    pub type ConsensusBlockHash<T: Config> =
```

Comment:
I think there is a another optimization that can done here using reference counting for all the domains rather than storing the same hash for each domain.

Reply:
I have considered this but it seems more complected because each domain may submit bundle in different consensus blocks so the consensus hash needed to store is different from domain to domain, and when should a consensus hash be pruned is also different from domain to domain, the complexity come when we need to track these difference.